source/Octopus.Tentacle/Kubernetes/KubernetesPodMonitor.cs

```diff
@@ -131,39 +152,68 @@ internal async Task OnNewEvent(WatchEventType type, V1Pod pod)
             }
         }
 
+        IList<PodStatus> IKubernetesPodStatusProvider.GetAllPodStatuses()
+        {
+            lookupLock.Wait();
+            try
+            {
+                return podStatusLookup.Values.ToList();
+            }
+            finally
+            {
+                lookupLock.Release();
+            }
```

Comment:
What if we made the `podStatusLookup` a `ConcurrentDictionary`? Could we avoid all the locking here?

Reply:
I did that first, but `InitialLoadAsync` gets re-called every 10 mins, so we need to make sure that the OrphanedPodCleaner doesn't call in while `podStatusLookup` is being refilled with data.

Reply:
hrm, ok, we can live with this for now :)

Reply:
I may have a solution that uses ConcurrentDictionary 👍 I'm just testing it.